api-js/src/queries/domains/find-domain-by-slug.js

```diff
@@ -4,14 +4,54 @@ const { domainType } = require('../../types')
 
 const findDomainBySlug = {
   type: domainType,
-  description: 'Select information on a specific domain using a slug.',
+  description: 'Retrieve a specific domain by providing a slug.',
   args: {
     urlSlug: {
       type: GraphQLNonNull(Slug),
-      description: 'The slugified domain you want to retrieve data for.',
+      description: 'The slugified name of the domain you wish to retrieve.',
     },
   },
-  resolve: async () => {},
+  resolve: async (
+    _,
+    args,
+    {
+      userKey,
+      query,
+      auth: { checkDomainPermission, userRequired },
+      loaders: { domainLoaderBySlug, userLoaderByKey },
+      validators: { cleanseInput },
+    },
+  ) => {
+    // Cleanse input
+    const urlSlug = cleanseInput(args.urlSlug)
+
+    // Get User
+    const user = await userRequired(userKey, userLoaderByKey)
+
+    // Retrieve domain by slug
+    const domain = await domainLoaderBySlug.load(urlSlug)
+
+    if (typeof domain === 'undefined') {
+      console.warn(`Could not retrieve domain.`)
+      throw new Error(`No domain with the provided slug could be found.`)
+    }
+
+    // Check user permission for domain access
+    const permitted = await checkDomainPermission(user._id, domain._id, query)
+
+    if (!permitted) {
+      console.warn(`User ${user._key} not permitted to access domain.`)
+      throw new Error(
+        `User ${user._key} is not permitted to access specified domain.`,
```

Comment:
Maybe make this a little more ambiguous just so that the user/attacker does not know that the domain does exist already